rust/src/storage/file/mod.rs

```diff
@@ -92,10 +92,16 @@ impl StorageBackend for FileStorageBackend {
         let readdir = ReadDirStream::new(fs::read_dir(path).await?);
 
         Ok(Box::pin(readdir.err_into().and_then(|entry| async move {
-            Ok(ObjectMeta {
-                path: String::from(entry.path().to_str().unwrap()),
-                modified: DateTime::from(entry.metadata().await.unwrap().modified().unwrap()),
-            })
+            let path = String::from(entry.path().to_str().unwrap());
+            // We check if the file is a temporary file created to prepare a commit.
+            // Getting meta data is another system call on some platforms and the file
+            // may have been removed in the meantime ...
+            let modified = if path.contains("_commit_") {
+                chrono::Utc::now()
+            } else {
+                DateTime::from(entry.metadata().await.unwrap().modified().unwrap())
+            };
```

Comment:
I'm not really happy with this solution, especially since we cannot do this if we are to move to an external object store crate. I toyed around with using a separate "directory" for preparing the commit files, to not have ephemeral files show up in the list operation at all, but there some errors popped up, as it seems due to renaming into a folder that does not exist yet, when creating a new table.

As is all concurrent tests are passing, so I thought I'd ask other peoples opinions.

FYI - the implementation in object_store_rs is quite different, and this problem might not show up there in that form, since there are now "unwraps" over there...

Reply:
Perhaps a more general logic (that would be acceptable in object_store_rs) would be to skip the entry if grabbing metadata resulted in `NotFound`?